### PR TITLE
(WIP) Add desktops to Developer-Guide_Build-Options.md

### PR DESCRIPTION
**Pull request by RichNeese**
adding functions for new desktop builder Functions

### CODE REVIEW

**Comment by igorpecovnik** [comment]
OK, let's build this up here, but not merge until we are ready.

**Comment by EvilOlaf** [comment]
@RichNeese can you rebase on master?

File: docs/Developer-Guide_Build-Options.md [hunk 1]
```diff
@@ -63,12 +63,19 @@ These parameters are meant to be applied to the `./compile.sh` command. They are
 - **BOARD** ( `string` ): set name of board manually to skip dialog prompt
 - **BRANCH** ( legacy | current | dev ): set kernel and U-Boot branch manually to skip dialog prompt; some options may not be available for all devices
 - **RELEASE** ( stretch | buster | bionic | focal | bullseye ): set OS release manually to skip dialog prompt; use this option with `KERNEL_ONLY=yes` to create board support package  
-  
+
+## Hidden options to minimize user input for desktop build automation
+- **DESKTOP_ENVIRONMENT** ( `Xfce` | `Budgie` | `Cinnamon` | `DDE` | `Enlightenment` | `Gnome3` | `Lxde` | `Mate` )
+- **DESKTOP_ENVIRONMENT_CONFIG_NAME** ( `config_basic` | `config_full` ): select for minimal no frills desktop or for full functional desktop
+- **DESKTOP_APPGROUPS_SELECTED** ( `browsers chat desktop_tools editors email internet multimedia office programming remote_desktop` ): Select what application groups to add to build
+- **DESKTOP_APT_FLAGS_SELECTED** ( `recommends suggests` ): Select to install suggested and/or recommended addons
```
Comment (igorpecovnik) [inline docs/Developer-Guide_Build-Options.md:71]: We have removed this one.

File: docs/Developer-Guide_Build-Options.md [hunk 2]
```diff
@@ -63,12 +63,19 @@ These parameters are meant to be applied to the `./compile.sh` command. They are
 - **BOARD** ( `string` ): set name of board manually to skip dialog prompt
 - **BRANCH** ( legacy | current | dev ): set kernel and U-Boot branch manually to skip dialog prompt; some options may not be available for all devices
 - **RELEASE** ( stretch | buster | bionic | focal | bullseye ): set OS release manually to skip dialog prompt; use this option with `KERNEL_ONLY=yes` to create board support package  
-  
+
+## Hidden options to minimize user input for desktop build automation
+- **DESKTOP_ENVIRONMENT** ( `Xfce` | `Budgie` | `Cinnamon` | `DDE` | `Enlightenment` | `Gnome3` | `Lxde` | `Mate` )
+- **DESKTOP_ENVIRONMENT_CONFIG_NAME** ( `config_basic` | `config_full` ): select for minimal no frills desktop or for full functional desktop
+- **DESKTOP_APPGROUPS_SELECTED** ( `browsers chat desktop_tools editors email internet multimedia office programming remote_desktop` ): Select what application groups to add to build
```
Comment (igorpecovnik) [inline docs/Developer-Guide_Build-Options.md:70]: 3dsupport

**Pull request closed without merging**